src/libAtomVM/term.h

```diff
@@ -773,13 +828,7 @@ static inline term term_from_const_binary(const void *data, uint32_t size, Conte
 */
 static inline term term_create_uninitialized_binary(uint32_t size, Context *ctx)
 {
-    int size_in_terms = term_binary_data_size_in_terms(size);
-
-    term *boxed_value = memory_heap_alloc(ctx, size_in_terms + 1);
-    boxed_value[0] = (size_in_terms << 6) | 0x24; // heap binary
-    boxed_value[1] = size;
-
-    return ((term) boxed_value) | TERM_BOXED_VALUE_TAG;
+    return term_from_literal_binary(NULL, size, ctx);
```

Comment:
This change is quite weird, we should use `term_create_uninitialized_binary` as a building block for `term_from_literal_binary` instead of viceversa.

Reply:
I like and made the changes you suggested.  It likely makes the diffs a little more complicated, because I had to move some functions around in order to avoid having to write forward declarations, which I thought was overkill, since it is not needed.  I hope the diffs are easy to read.